### ISSUE TITLE
Updating marker position with change in Lat-Lng position of marker in Google Maps Flutter Plugin?

### ISSUE THREAD

**Issue by Maverick099** [flutter/flutter#54515]
I am making a vehicle tracking administration app using flutter. I have implemented the maps in the app using the ```google-maps-flutter``` plugin. **The part I am stuck is how to update the marker position when there is a change in the location of the vehicle**.  My location data of the vehicles are stored in the firestore database which I can get into the app using the ```StreamBuilder ```.

I found a ```updateMarker``` function inside the plugin code but don't know how to implement it.
![stackoverflow question](https://user-images.githubusercontent.com/32545664/79034093-fb6f2280-7bd0-11ea-83df-4ff644797fff.jpg)

If I can not ask this type of here: I have asked the same problem in StackOverflow too the link is here:[https://stackoverflow.com/questions/61136968/updating-marker-position-with-change-in-lat-lng-position-of-marker-in-google-map]( https://stackoverflow.com/questions/61136968/updating-marker-position-with-change-in-lat-lng-position-of-marker-in-google-map)

**Comment by imeDevelopers** [comment]
i am working on similar project here in Middle East ...

i will tell you the idea without code ..

you can simply create a list of Marker like:

List<Marker> myMarkers = List<Marker>();

and add the markers to it , if there a new data or new position .. clear it and add the new markers to it again and pass if to the map by using Set<Marker>.from(myMarkers) in the (markers) parameter of the map ...


hope it helps ..

**Comment by Maverick099** [comment]
@imeDevelopers yeah, this was my first approach to the problem and call the set state after an interval of time like for every 5 sec but when I saw the update marker code I thought there is an efficient way to do it.

And I had a doubt on using the method this `setState` method that you have implemented, by calling the `setState`  so many time does the whole map load again in the process of updating the markers or just the markers change their state? because if the whole map loads again then the application is not efficient right?

It would be very helpful if you can clear this doubt that I have and thank you for the response.

**Comment by imeDevelopers** [comment]
there is no any problems with so many setstate .. , the map will not load again .. its const widget .. the markers only are updated with setstate ..

**Comment by Maverick099** [comment]
@imeDevelopers can you help me with the SetSate() for updating the markers.

**Comment by jweidner-mbible** [comment]
@Maverick099 In dart, if an identifier starts with an underscore (_), it’s private to its library.  So the _updateMarkers method isn't something you can call.

**Comment by Maverick099** [comment]
@jweidner-mbible ok I get it now .

**Comment by Maverick099** [comment]
I am facing the issue where after I updated the Markers List and called the setState the position of the map is not updating. Is there another way to do it? @imeDevelopers @TahaTesser

**Comment by Maverick099** [comment]
I found a way to update the position of the markers using the `MarkerUpdates` class. The same class is mentioned in the documentation of the Google-Maps-Plugin. This class takes two  `Set<Marker>` as inputs one the current Markers Set and another the new updated Marker Set.  The documentation for this class is here: https://pub.dev/documentation/google_maps_flutter_platform_interface/latest/google_maps_flutter_platform_interface/MarkerUpdates-class.html


To use this class you will have to add this import statement :
`import 'package:google_maps_flutter_platform_interface/src/types/marker_updates.dart';`

**While performing this method my google-maps plugin version was `google_maps_flutter: ^0.5.29+1`**

### then make a function as follows :
```
List<Markers> markers; //This the list of markers is the old set of markers that were used in the onMapCreated function 

void upDateMarkers() {
  List<Markers> updatedMarkers =[]; //new markers with updated position go here 

  updatedMarkers =['updated the markers location here and also other properties you need.'];
  

  /// Then call the SetState function.
  /// I called the MarkersUpdate class inside the setState function.
  /// You can do it your way but remember tp call the setState function so that the updated markers reflect on your Flutter app.
  /// Ps: I did not try the second way where the MarkerUpdate is called outside the setState buttechnically it should work.
  setState(() {
    MarkerUpdates.from(
        Set<Marker>.from(markers), Set<Marker>.from(updatedMarkers));
    markers = [];
    markers = updatedMarkers;
 //swap of markers so that on next marker update the previous marker would be the one which you updated now.
// And even on the next app startup it takes the updated markers to show on the map.
  });
}

```

then call the function periodically in my case or as you wish the markers will update.

While doing this  a warning as I was promoted with a warning as :
`Don't import implementation files from another package.dartimplementation_imports`

I don't know if it's a safe approach to do but it is doing the job.
It would be great if @TahaTesser or any other form the @flutter or google-maps-flutter team could tell us more about the warning if it has a potential of creating a threat.

And it would very nice if devs made this class able to import from the main google maps plugin package instead from an implementation file or may I create a pull request by updating the main plugin file.

Note:

There is a similar class to update the circle, polygons and options (Map Options) the documentation has explained all and imports for those classes are similar in the same path as mentioned for the `Updatemarkers` class.

**Comment by sugitatestblue** [comment]
is it fix?
